File: backend/SKyber/src/main/java/edu/cit/SKyber/Entity/SKProfile.java
package edu.cit.SKyber.Entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class SKProfile {

    @Id
    private String uid; // Use UID as the primary key instead of auto-incremented id
    private String firstName;
    private String lastName;
    private String email;
    private String position;
    private String term; //ex 2015-2018
    private String platform;
    private String birthdate;
    private String gender;
    private int age;
    private String phoneNumber;
    private String address;
    private String skImage;
    private String role="ADMIN"; // Added role field for user roles (Optional)

}

File: frontend_mobile/app/src/main/java/com/example/skyber/dataclass/SKProfile.kt
package com.example.skyber.dataclass

import android.os.Parcelable
import com.google.firebase.database.IgnoreExtraProperties
import kotlinx.parcelize.Parcelize

@Parcelize
@IgnoreExtraProperties
data class SKProfile(
    val uid: String? = "",
    val firstName: String? = "",
    val lastName: String? = "",
    val email: String? = "",
    val position: String? = "",
    val term: String? = "",       // e.g., "2015-2018"
    val platform: String? = "",
    val birthdate: String? = "",
    val gender: String? = "",
    val age: Int = 0,
    val phoneNumber: String? = "",
    val address: String? = "",
    val skImage: String? = null, // Base64 string or URL
    val role: String = "ADMIN"   // Default role
) : Parcelable